internal/pass/sync/handlers.go

```diff
@@ -0,0 +1,141 @@
+package sync
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/twofas/2fas-server/internal/common/logging"
+	"github.com/twofas/2fas-server/internal/pass/connection"
+)
+
+func ExtensionRequestSync(syncingApp *Syncing) gin.HandlerFunc {
+	return func(gCtx *gin.Context) {
+		token, err := connection.TokenFromWSProtocol(gCtx.Request)
+		if err != nil {
+			logging.Errorf("Failed to get token from request: %v", err)
+			gCtx.Status(http.StatusForbidden)
+			return
+		}
+		fcmToken, err := syncingApp.VerifyExtRequestSyncToken(gCtx, token)
+		if err != nil {
+			logging.Errorf("Failed to verify proxy token: %v", err)
+			gCtx.String(http.StatusUnauthorized, "Invalid auth token")
+			return
+		}
+
+		if err := syncingApp.ServeSyncingRequestWS(gCtx.Writer, gCtx.Request, fcmToken); err != nil {
+			logging.Errorf("Failed to verify proxy token: %v", err)
+			gCtx.Status(http.StatusInternalServerError)
```

Comment:
have you checked if setting status here after connection.Upgrade works? I don't think so.

Reply:
`ServeSyncingRequestWS` will not return error after upgrade. It will return error if upgrade fails or if there is problem with the protocol in the headers.